### PR TITLE
[spaceship] Add exception handling and guidance when loading teams

### PR DESCRIPTION
**Pull request by bgannin**
Crash: When no teams exist in your AppleID account #2003

Wrapped team selection in exception handling with informative text about what might have gone wrong and how to fix it. 'rake test_all' passes with no failures or issues.

### CODE REVIEW

**Comment by KrauseFx** [comment]
Thanks for the pull request @bgannin, however we'd have to add this to every tool that uses spaceship. Instead, could we add this to `spaceship` itself maybe? However in `spaceship` we try to print as little as possible, but we do have an UI layer there. What do you think?

**Comment by bgannin** [comment]
Sure, I'll take a look at that instead.

**Comment by KrauseFx** [comment]
Thanks @bgannin, please let me know if you're unsure or if there is any way we can help 👍

**Comment by bgannin** [comment]
Migrated it over, hopefully in the right fashion :) Thanks!

File: spaceship/lib/spaceship/portal/ui/select_team.rb [hunk 1]
```diff
@@ -32,9 +32,15 @@ class UserInterface
       #   ]
 
       def select_team
-        teams = client.teams
-
-        raise "Your account is in no teams" if teams.count == 0
+        begin
+          teams = client.teams
+          raise "Your account is in no teams" if teams.count == 0
```
Comment (KrauseFx) [inline spaceship/lib/spaceship/portal/ui/select_team.rb]: Why do we first raise the error, just to catch it and swallow it?

**Comment by KrauseFx** [comment]
This is great, thanks 👍 I added 2 comments. I like the text + link

File: spaceship/lib/spaceship/portal/ui/select_team.rb [hunk 2]
```diff
@@ -32,9 +32,15 @@ class UserInterface
...
+        rescue
+          puts "No teams available on the Developer Portal"
+          puts "⚠️ you must accept an invitation to a team for it to be available"
+          puts "To learn more about teams and how to use them visit https://developer.apple.com/library/ios/documentation/IDEs/Conceptual/AppDistributionGuide/ManagingYourTeam/ManagingYourTeam.html"
+          return
```
Comment (bgannin) [inline spaceship/lib/spaceship/portal/ui/select_team.rb]: I'm pretty fresh to Ruby, feel free to chalk it up to that. Would re-raising the exception cause the match command crash this came from again?

File: spaceship/lib/spaceship/portal/ui/select_team.rb (hunk 1 above)
Comment (bgannin) [inline spaceship/lib/spaceship/portal/ui/select_team.rb]: My thinking:

* client.teams can raise an exception on its own (the original crash) and…
* teams.count == 0 is an exception of the same variety

Check and catch both here with the appropriate rescue help text.

**Comment by KrauseFx** [comment]
I just pushed https://github.com/fastlane/fastlane/pull/5713, thanks for your work @bgannin 👍

**Pull request closed without merging**